cirq/sim/simulator.py

```diff
@@ -526,7 +525,51 @@ def _base_iterator(
         Yields:
             StepResults from simulating a Moment of the Circuit.
         """
-        raise NotImplementedError()
+        qubits = ops.QubitOrder.as_qubit_order(qubit_order).order_for(circuit.all_qubits())
+        acton_args = self._create_act_on_args(initial_state, qubits)
+        return self._core_iterator(circuit, acton_args, qubits)
+
+    @abc.abstractmethod
+    def _create_act_on_args(
```

Comment:
New abstract method would break any external simulators. Have to make this just a regular function returning NotImplemented.